fil_actor_interface/src/builtin/reward/mod.rs

```diff
@@ -100,4 +106,52 @@ impl State {
             State::V13(st) => from_token_v4_to_v2(st.into_total_storage_power_reward()),
         }
     }
+
+    /// The baseline power the network is targeting at this state's epoch.
+    pub fn this_epoch_baseline_power(self) -> StoragePower {
+        match self {
+            State::V8(st) => st.this_epoch_baseline_power,
+            State::V9(st) => st.this_epoch_baseline_power,
+            State::V10(st) => st.this_epoch_baseline_power,
+            State::V11(st) => st.this_epoch_baseline_power,
+            State::V12(st) => st.this_epoch_baseline_power,
+            State::V13(st) => st.this_epoch_baseline_power,
+        }
+    }
+
+    pub fn deal_provider_collateral_bounds(
+        size: PaddedPieceSize,
+        raw_byte_power: StoragePower,
+        baseline_power: StoragePower,
+        network_circulating_supply: TokenAmount,
+    ) -> (TokenAmount, TokenAmount) {
+        // The percentage of normalized circulating supply that must be covered by provider
+        // collateral in a deal.
+        // See: https://github.com/filecoin-project/go-state-types/blob/master/builtin/v12/market/policy.go#L9-L14.
+        // Note that as of v13 the logic is exactly the same (copy-paste) for all the versions.
+        // Should that change - the code here is going to need some adjustment.
+        let numerator = BigInt::new(Sign::Plus, vec![1]);
+        let denominator = BigInt::new(Sign::Plus, vec![100]);
```

Comment:
This is an extremely explicit way of defining `BigInt`s. Is there a particular reason `BigInt::from(1)` would not be sufficient here?

Reply:
Also, the exact values are defined in the policy and should not be re-defined. https://github.com/ChainSafe/fil-actor-states/blob/ec1cdbbf378156c59450d8ba891ac4d64d0b0303/fil_actors_shared/src/v12/runtime/policy.rs#L347-L349

Reply:
Missed this, there's quite a bit of a difference compared to Lotus code.